listener/upload.go

```diff
@@ -97,9 +97,10 @@ func HandleUpload(event HostEvent) error {
 			return nil
 		}
 	}
-
+	t := time.Now().Format(base.Rfc3339NoTz)
```

Comment:
Have you tried to use time.Time type for PlatformEvent.Timestamp field? Maybe it would ser/deserialize it out of the box.

Reply:
If the field is missing, then it will be deserialized to zero type - which is weird to work with in the case of `time.Time`. Pointer is much clearer

Reply:
@semtexzv you can use `*time.Time` so it should be nil once field is not presented.

Reply:
Resolved by creating a wrapper which works with required datetime format.

Reply:
Ok, but I don't see it. Still see string ser/deser.